### PR TITLE
Fix false positives in inclusive_language rule

### LINKED ISSUE #3415: inclusive_language false positives: Triggering terms as substrings of identifiers
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

We have a project that deals with payments provided by Mastercard and as such it is common to have code such as:

`let mastercard = newWalletItem(for: .mastercard)`

This triggers a violation of the `inclusive_language` rule as the identifier contains a triggering term as a substring.

Is it possible to the break triggering terms in identifiers using camel case so `let masterThread` is a violation but `let mastercard` is not?

##### Complete output when running SwiftLint, including the stack trace and command used

N/A

### Environment

* SwiftLint version: 0.41.0
* Installation method used: CocoaPods
* Paste your configuration file:
N/A - Default rule

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  N/A
* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 12.1

* Do you have a sample that shows the issue?

```swift
// This triggers a violation:
`let mastercard = newWalletItem(for: .mastercard)`
```

```swift
public enum WalletItemType: String {
    case visa = "VISA"
    // This triggers a violation:
    case mastercard = "MASTERCARD"
}
```

### PR DESCRIPTION
**Pull request by daltonclaybrook**
This PR resolves #3415 by adding "allowed terms" to the rule configuration. If a disallowed term is found to overlap with an allowed term, the allowed term "wins." The only default allowed term at the moment is "mastercard"

### CODE REVIEW

**Comment by daltonclaybrook** [comment]
@jpsim let me know what you think of this approach vs the old camel-case/regex strategy (#3422). I think this is a lot simpler.

**Comment by jpsim** [comment]
Yes this is a lot simpler. Will take a closer look.

File: CHANGELOG.md [hunk 1]
```diff
@@ -74,7 +74,9 @@
 
 #### Bug Fixes
 
-* None.
+* Fix false positives in `inclusive_language` rule.
```
Comment (jpsim) [inline CHANGELOG.md:77]: Technically, "mastercard" wasn't a false positive but rather just a commonly used name that we're special-casing it here. I'd also say this is less of a bug fix and more of an enhancement.

I'll push up some changes to the changelog to reflect this.

**Review by jpsim — APPROVED** [review]
Thanks for addressing this! 🙌

**Comment by apps4everyone** [comment]
is there a way to also allow UPPERCASED version in the swiftlint.yml?
e.g: case masterCard = "MASTERCARD"

**Pull request merged**